core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -800,12 +800,15 @@ class DynamicMetricsReporters(brokerId: Int, server: KafkaBroker) extends Reconf
     updatedConfigs.forEach { (k, v) => props.put(k, v.asInstanceOf[AnyRef]) }
     propsOverride.forKeyValue { (k, v) => props.put(k, v) }
     val reporters = dynamicConfig.currentKafkaConfig.getConfiguredInstances(reporterClasses, classOf[MetricsReporter], props)
+    // call notifyMetricsReporters first to satisfy the contract for MetricsReporter.contextChange,
+    // which provides that MetricsReporter.contextChange must be called before the first call to MetricsReporter.init().
+    // The first call to init() is done when we call metrics.addReporter below.
```

Comment:
nit: `call` -> `Call`. Why did you put `()` for `MetricsReporter.init()` but not for `notifyMetricsReporters`,  `MetricsReporter.contextChange` and `metrics.addReporter`? We could perhaps remove the `()` to stay consistent.

Reply:
done